src/modules/claim/claim.service.ts

```diff
@@ -379,4 +362,33 @@ export class ClaimService {
     const { sub: subject } = jwt.decode(token);
     return v5(JSON.stringify({ subject, role, version }), UUID_NAMESPACE);
   }
+
+  private async rolesByIssuer(issuer: string, namespace?: string): Promise<Role[]> {
+    const rolesOfIssuer = (await this.getBySubject({ subject: issuer, filters: { isAccepted: true } }))
+      .map((r) => r.claimType);
+    const roles = (await this.roleService.getAll())
+      .filter(
+        (r) => r.definition.issuer.did?.includes(issuer) || rolesOfIssuer.includes(r.definition.issuer.roleName)
+      );
+    return namespace ? roles.filter((r: Role) => r.namespace === namespace) : roles;
+  }
+
+  private async filterUserRelatedClaims(currentUser: string, qb: SelectQueryBuilder<Claim>) {
+    const ownedAssets = (await this.assetService.getByOwner(currentUser)).map((a) => a.id);
+    const offeredAssets = (await this.assetService.getByOfferedTo(currentUser)).map((a) => a.id);
+
+    const rolesByUser = (await this.rolesByIssuer(currentUser)).map((r) => r.namespace);
+
+    qb.andWhere(new Brackets(query => {
+      query.where('claim.claimType IN (:...rolesByUser)', { rolesByUser })
+        .orWhere('claim.subject = :currentUser', { currentUser })
+        .orWhere('claim.requester = :currentUser ', { currentUser });
```

Comment:
Possibly a minor point, but maybe remove the space here. Not sure if it would cause an issue or not.
```suggestion
        .orWhere('claim.requester = :currentUser', { currentUser });
```